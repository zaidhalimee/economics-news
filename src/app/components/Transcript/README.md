## Description

The `Transcript` component renders video transcripts.

## Props

| Name       | type   | Description                 |
| ---------- | ------ | --------------------------- |
| transcript | object | contains transcript content |
| title      | string | title of video              |

## Example

```tsx
<Transcript transcript={transcriptBlock} title={mediaTitle} hideDisclaimer />
```
