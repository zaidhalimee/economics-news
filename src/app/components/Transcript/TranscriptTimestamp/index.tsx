/** @jsx jsx */
import { jsx } from '@emotion/react';
import styles from './index.styles';

// using span not time element to prevent text splitting bug on Talkback
const TranscriptTimestamp = ({ timestamp }: { timestamp: string }) => {
  return <span css={styles.time}>{timestamp}</span>;
};

export default TranscriptTimestamp;
