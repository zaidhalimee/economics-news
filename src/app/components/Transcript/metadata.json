{
  "lastUpdated": {
    "day": 28,
    "month": "February",
    "year": 202
  },
  "uxAccessibilityDoc": {
    "done": true,
    "reference": {
      "url": "https://paper.dropbox.com/doc/Transcript-first-video-Screen-reader-UX--CgiT6SE5mU5yG~eks1gzpEwoAg-ZJGBJjMxHv3RqSF2WbWAs",
      "label": "Screen Reader UX"
    }
  },
  "acceptanceCriteria": {
    "done": true,
    "reference": {
      "url": "https://paper.dropbox.com/doc/Transcript-first-video-Acceptance-criteria-and-Translations--ChOLnyw2l1ClkyYQ8XSxUVszAg-3dInEQq6OXzBLMvnrOqSa",
      "label": "Accessibility Acceptance Criteria"
    }
  },
  "swarm": {
    "done": false,
    "reference": {
      "url": "",
      "label": "Accessibility Swarm"
    }
  }
}
