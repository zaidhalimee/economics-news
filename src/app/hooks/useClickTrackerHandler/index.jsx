/* eslint-disable no-console */
import { useContext, useCallback, useState } from 'react';

import { EventTrackingContext } from '../../contexts/EventTrackingContext';
import useTrackingToggle from '../useTrackingToggle';
import OPTIMIZELY_CONFIG from '../../lib/config/optimizely';
import { sendEventBeacon } from '../../components/ATIAnalytics/beacon/index';
import { ServiceContext } from '../../contexts/ServiceContext';
import { isValidClick } from './clickTypes';

const EVENT_TYPE = 'click';

const useClickTrackerHandler = (props = {}) => {
  const preventNavigation = props?.preventNavigation;
  const componentName = props?.componentName;
  const url = props?.url;
  const advertiserID = props?.advertiserID;
  const format = props?.format;
  const optimizely = props?.optimizely;
  const optimizelyMetricNameOverride = props?.optimizelyMetricNameOverride;
  const detailedPlacement = props?.detailedPlacement;

  const { trackingIsEnabled } = useTrackingToggle(componentName);
  const [clicked, setClicked] = useState(false);
  const eventTrackingContext = useContext(EventTrackingContext);

  const { pageIdentifier, platform, producerId, statsDestination } =
    eventTrackingContext;
  const campaignID = props?.campaignID || eventTrackingContext?.campaignID;

  const { service } = useContext(ServiceContext);

  return useCallback(
    async event => {
      const shouldRegisterClick = [
        trackingIsEnabled,
        !clicked,
        isValidClick(event),
      ].every(Boolean);
      if (shouldRegisterClick) {
        setClicked(true);

        const shouldSendEvent = [
          campaignID,
          componentName,
          pageIdentifier,
          platform,
          producerId,
          service,
          statsDestination,
        ].every(Boolean);
        if (shouldSendEvent) {
          const nextPageUrl = event?.currentTarget?.href;

          event.stopPropagation();
          event.preventDefault();

          if (optimizely) {
            const eventName = OPTIMIZELY_CONFIG.viewClickAttributeId;

            const overrideAttributes = {
              ...optimizely.user.attributes,
              [`clicked_${eventName}`]: true,
            };

            optimizely.track(
              optimizelyMetricNameOverride
                ? `${optimizelyMetricNameOverride}_clicks`
                : 'component_clicks',
              optimizely.user.id,
              overrideAttributes,
            );
          }

          const optimizelyVariation =
            optimizely?.getVariation(OPTIMIZELY_CONFIG.ruleKey) || null;

          try {
            await sendEventBeacon({
              type: EVENT_TYPE,
              campaignID,
              componentName,
              format,
              pageIdentifier,
              platform,
              producerId,
              service,
              advertiserID,
              statsDestination,
              url,
              detailedPlacement,
              ...(optimizelyVariation &&
                optimizelyVariation !== 'off' && {
                  experimentVariant: optimizelyVariation,
                }),
            });
          } finally {
            if (nextPageUrl && !preventNavigation) {
              if (optimizely) {
                optimizely.close();
              }
              window.location.assign(nextPageUrl);
            }
          }
        }
      }
    },
    [
      trackingIsEnabled,
      clicked,
      campaignID,
      componentName,
      pageIdentifier,
      platform,
      preventNavigation,
      producerId,
      service,
      statsDestination,
      url,
      advertiserID,
      format,
      optimizely,
      optimizelyMetricNameOverride,
      detailedPlacement,
    ],
  );
};

export default useClickTrackerHandler;
