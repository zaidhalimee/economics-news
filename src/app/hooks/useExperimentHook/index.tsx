import useOperaMiniDetection from '#app/hooks/useOperaMiniDetection';
import { Services } from '#app/models/types/global';
import { useContext, useEffect, useState } from 'react';
import { ServiceContext } from '../../contexts/ServiceContext';

// Disabled due to bug in ts lint
// eslint-disable-next-line no-shadow
export enum Stages {
  STAGE_1 = 'stage_1',
  STAGE_2 = 'stage_2',
  STAGE_3 = 'stage_3',
  DEFAULT = 'default',
}

type ExperimentCriteria = Partial<{
  service: Services;
  isOperaMini: boolean;
  dataSaver: boolean;
  lowPower: boolean;
  noJs: boolean;
}>;

export type Navigator = {
  connection: { saveData: boolean };
  getBattery?: () => Promise<{ level: number }>;
};

const LOW_POWER_THRESHOLD = 0.2;

const determineStage = ({
  service,
  isOperaMini,
  dataSaver,
  lowPower,
  noJs,
}: ExperimentCriteria) => {
  if (noJs) {
    return Stages.STAGE_1;
  }

  if (service !== 'mundo' && !lowPower && !dataSaver && !isOperaMini) {
    return Stages.STAGE_3;
  }

  if (service === 'mundo' || dataSaver || isOperaMini || lowPower) {
    return Stages.STAGE_2;
  }

  return Stages.DEFAULT;
};

const useExperimentHook = () => {
  const [lowPower, setLowPower] = useState(false);
  const [dataSaver, setSaveDataMode] = useState(false);
  const [noJs, setNoJs] = useState(true);
  const isOperaMini = useOperaMiniDetection();
  const { service } = useContext(ServiceContext);

  useEffect(() => {
    const initialiseDeviceStates = async () => {
      const nav = navigator as unknown as Navigator;
      const saveDataMode = nav.connection?.saveData;
      if (nav.getBattery) {
        const manager = await nav.getBattery();
        const { level } = manager;
        const isLowPower = level <= LOW_POWER_THRESHOLD;
        setLowPower(isLowPower);
      }
      setSaveDataMode(saveDataMode);
    };
    initialiseDeviceStates();
    setNoJs(false);
  }, []);

  const stage = determineStage({
    isOperaMini,
    service,
    dataSaver,
    lowPower,
    noJs,
  });

  return stage;
};

export default useExperimentHook;
