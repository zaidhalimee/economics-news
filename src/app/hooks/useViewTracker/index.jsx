import { useContext, useEffect, useState, useRef } from 'react';
import prop from 'ramda/src/prop';

import { RequestContext } from '#app/contexts/RequestContext';
import { sendEventBeacon } from '../../components/ATIAnalytics/beacon';
import { EventTrackingContext } from '../../contexts/EventTrackingContext';
import useTrackingToggle from '../useTrackingToggle';
import OPTIMIZELY_CONFIG from '../../lib/config/optimizely';
import { ServiceContext } from '../../contexts/ServiceContext';
import { useConstructLiteSiteATIEventTrackUrl } from '../useClickTrackerHandler';

const VIEW_EVENT = 'view';
const VIEWED_DURATION_MS = 1000;
const MIN_VIEWED_PERCENT = 0.5;
export const LITE_ATI_VIEW_TRACKING = 'data-lite-ati-view-tracking';

/**
 *
 * @returns {Ref<HTMLElement> | undefined}
 */
const useViewTracker = (props = {}) => {
  const componentName = props?.componentName;
  const format = props?.format;
  const advertiserID = props?.advertiserID;
  const url = props?.url;
  const optimizely = props?.optimizely;
  const optimizelyMetricNameOverride = props?.optimizelyMetricNameOverride;
  const detailedPlacement = props?.detailedPlacement;

  const observer = useRef();
  const timer = useRef(null);
  const [isInView, setIsInView] = useState();
  const [eventSent, setEventSent] = useState(false);
  const { trackingIsEnabled } = useTrackingToggle(componentName);
  const eventTrackingContext = useContext(EventTrackingContext);

  const {
    pageIdentifier,
    platform,
    producerId,
    producerName,
    statsDestination,
  } = eventTrackingContext;
  const campaignID = props?.campaignID || eventTrackingContext?.campaignID;

  const { service, useReverb } = useContext(ServiceContext);

  const initObserver = async () => {
    if (typeof window.IntersectionObserver === 'undefined') {
      // Polyfill IntersectionObserver, e.g. for IE11
      await import('intersection-observer');
    }
    const callback = changes => {
      const someElementsAreInView = changes.some(prop('isIntersecting'));

      setIsInView(someElementsAreInView);
    };
    const options = {
      threshold: [MIN_VIEWED_PERCENT],
    };

    observer.current = new IntersectionObserver(callback, options);
  };

  useEffect(() => {
    if (isInView && !timer.current) {
      timer.current = setTimeout(() => {
        const hasRequiredProps = [
          campaignID,
          componentName,
          pageIdentifier,
          platform,
          producerId,
          producerName,
          service,
          statsDestination,
        ].every(Boolean);

        const shouldSendEvent = [
          hasRequiredProps,
          trackingIsEnabled,
          !eventSent,
        ].every(Boolean);

        if (shouldSendEvent) {
          if (optimizely) {
            const eventName = OPTIMIZELY_CONFIG.viewClickAttributeId;

            const overrideAttributes = {
              ...optimizely.user.attributes,
              [`viewed_${eventName}`]: true,
            };

            optimizely.track(
              optimizelyMetricNameOverride
                ? `${optimizelyMetricNameOverride}_views`
                : 'component_views',
              optimizely.user.id,
              overrideAttributes,
            );
          }

          const optimizelyVariation =
            optimizely?.getVariation(OPTIMIZELY_CONFIG.ruleKey) || null;

          sendEventBeacon({
            campaignID,
            componentName,
            format,
            pageIdentifier,
            platform,
            producerId,
            producerName,
            service,
            statsDestination,
            type: VIEW_EVENT,
            advertiserID,
            url,
            detailedPlacement,
            useReverb,
            ...(optimizelyVariation &&
              optimizelyVariation !== 'off' && {
                experimentVariant: optimizelyVariation,
              }),
          });
          setEventSent(true);
          observer.current.disconnect();
          observer.current = null;
          timer.current = null;
        }
      }, VIEWED_DURATION_MS);
    } else {
      clearTimeout(timer.current);
      timer.current = null;
    }

    return () => {
      clearTimeout(timer.current);
    };
  }, [
    campaignID,
    componentName,
    format,
    isInView,
    pageIdentifier,
    platform,
    producerId,
    producerName,
    service,
    statsDestination,
    trackingIsEnabled,
    eventSent,
    advertiserID,
    url,
    optimizely,
    optimizelyMetricNameOverride,
    detailedPlacement,
    useReverb,
  ]);

  return async element => {
    if (!element || !trackingIsEnabled || eventSent) {
      return;
    }
    if (!observer.current) {
      await initObserver();
    }

    observer.current.observe(element);
  };
};

export const useLiteViewTracker = (props = {}) => {
  const { isLite } = useContext(RequestContext);
  const liteHandler = useConstructLiteSiteATIEventTrackUrl({
    props,
    eventType: VIEW_EVENT,
  });

  return isLite ? { [LITE_ATI_VIEW_TRACKING]: liteHandler } : null;
};

export default useViewTracker;
