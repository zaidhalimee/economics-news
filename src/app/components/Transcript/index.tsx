/** @jsx jsx */
/* eslint-disable jsx-a11y/aria-role */
import { jsx } from '@emotion/react';
import useViewTracker from '#app/hooks/useViewTracker';
import styles from './index.styles';
import Text from '../Text';
import TranscriptTimestamp from './TranscriptTimestamp';
import VisuallyHiddenText from '../VisuallyHiddenText';
import { RightArrow as ArrowSvg } from '../icons';
import { TranscriptBlock, TranscriptItem } from './types';

// TO DO - move this to BFF
const removeHoursMilliseconds = (timestamp: string) => timestamp.slice(3, -4);

const TranscriptListItem = ({ id, start, content }: TranscriptItem) => (
  <li key={id} css={styles.listItem}>
    <Text role="text" css={styles.transcriptText} size="bodyCopy">
      <TranscriptTimestamp timestamp={removeHoursMilliseconds(start)} />
      <VisuallyHiddenText> </VisuallyHiddenText>
      <span css={styles.itemText}>{content}</span>
    </Text>
  </li>
);

const Transcript = ({
  transcript,
  title,
}: {
  transcript: TranscriptBlock;
  title?: string;
}) => {
  const viewRef = useViewTracker({ componentName: 'Transcript' });
  const transcriptItems = transcript?.model?.blocks;
  if (!transcriptItems) {
    return null;
  }

  const formattedTitle = title ? `, ${title}` : '';

  return (
    <details css={styles.details}>
      <summary css={styles.summary}>
        <ArrowSvg />
        <span role="text">
          {/* TO DO - add translations */}
          <Text size="pica" fontVariant="sansBold" css={styles.summaryTitle}>
            Read transcript
          </Text>
          {title && <VisuallyHiddenText>{formattedTitle}</VisuallyHiddenText>}
        </span>
      </summary>
      <ul css={styles.ul} role="list" ref={viewRef}>
        {/*  eslint-disable-next-line @typescript-eslint/no-unused-vars */}
        {transcriptItems.map((item, _index) => (
          <TranscriptListItem
            key={item.id}
            id={item.id}
            start={item.start}
            content={item.content}
          />
        ))}
      </ul>
      <Text size="brevier" css={styles.disclaimer} as="small">
        This transcript was reviewed by a journalist after AI generation.
      </Text>
    </details>
  );
};

export default Transcript;
