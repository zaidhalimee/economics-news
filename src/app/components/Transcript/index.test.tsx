import React from 'react';
import { render } from '../react-testing-library-with-providers';
import transcriptFixture from './fixture.json';
import Transcript from './index';
import * as viewTracking from '../../hooks/useViewTracker';

describe('Transcript Component', () => {
  it('should match snapshot (temp)', () => {
    const { container } = render(
      <Transcript transcript={transcriptFixture} title="My Title" />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render details element', () => {
    const { container } = render(
      <Transcript transcript={transcriptFixture} title="My Title" />,
    );
    const details = container.querySelector('details');
    expect(details).toBeInTheDocument();
  });

  it('should render summary element', () => {
    const { container } = render(<Transcript transcript={transcriptFixture} />);
    const summary = container.querySelector('summary');
    expect(summary).toBeInTheDocument();
  });

  it('should render the title as a visually hidden element inside the summary', () => {
    const { container } = render(
      <Transcript transcript={transcriptFixture} title="My Title" />,
    );
    const summaryWithTitle = container.querySelector('summary');
    expect(summaryWithTitle).toHaveTextContent('Read transcript, My Title');
  });

  it('should render an unordered list element with role list', () => {
    const { container } = render(
      <Transcript transcript={transcriptFixture} title="My Title" />,
    );
    const unorderedList = container.querySelector('ul');
    expect(unorderedList).toBeInTheDocument();
    expect(unorderedList).toHaveRole('list');
  });

  it('should render multiple list elements', () => {
    const { container } = render(
      <Transcript transcript={transcriptFixture} title="My Title" />,
    );
    const listItems = container.querySelectorAll('li');
    expect(listItems).toHaveLength(34);
  });

  it('should not render if there are no transcript items', () => {
    const { container } = render(
      // @ts-expect-error unexpected value
      <Transcript transcript={[]} title="My Title" />,
    );
    const details = container.querySelector('details');
    expect(details).not.toBeInTheDocument();
  });

  describe('view tracking', () => {
    const viewTrackerSpy = jest.spyOn(viewTracking, 'default');

    it('should register view tracking', () => {
      render(<Transcript transcript={transcriptFixture} title="My Title" />);

      expect(viewTrackerSpy).toHaveBeenCalledWith({
        componentName: 'Transcript',
      });
    });
  });
});
