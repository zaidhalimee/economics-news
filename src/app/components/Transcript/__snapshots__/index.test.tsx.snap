// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transcript Component should match snapshot (temp) 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  display: block;
  border: solid 0.1875rem transparent;
}

.emotion-0 summary svg {
  color: #141414;
  fill: currentcolor;
  width: 1rem;
  height: 1rem;
  vertical-align: text-top;
}

.emotion-0[open] summary svg {
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-1 {
  list-style: none;
  padding: 1rem 0.25rem;
}

.emotion-1::-webkit-details-marker {
  display: none;
}

.emotion-1:hover,
.emotion-1:focus {
  cursor: pointer;
}

.emotion-1:hover span,
.emotion-1:focus span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-1:focus-visible {
  outline: 0.1875rem solid #000000;
  outline-offset: -0.375rem;
}

.emotion-2 {
  color: #141414;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 700;
  color: #141414;
  -webkit-padding-start: 0.25rem;
  padding-inline-start: 0.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-3 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-4 {
  color: #141414;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  color: #545658;
  display: block;
  padding-bottom: 1rem;
  padding-inline: 0.5rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

.emotion-5 {
  padding: 0 0.5rem;
  list-style: none;
  margin: 0;
}

.emotion-6 {
  padding-bottom: 1rem;
}

.emotion-6::after {
  content: "";
  display: block;
  clear: both;
}

.emotion-7 {
  color: #141414;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  color: #545658;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-7 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-7 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

.emotion-8 {
  float: inline-start;
  width: 100%;
}

@media (min-width: 15rem) {
  .emotion-8 {
    width: auto;
  }
}

.emotion-10 {
  float: inline-start;
  width: 100%;
}

@media (min-width: 15rem) {
  .emotion-10 {
    -webkit-padding-start: 0.5rem;
    padding-inline-start: 0.5rem;
    width: calc(75% - 0.5rem);
  }
}

@media (min-width: 25rem) {
  .emotion-10 {
    width: calc(85% - 0.5rem);
  }
}

@media (min-width: 37.5rem) {
  .emotion-10 {
    -webkit-padding-start: 1rem;
    padding-inline-start: 1rem;
    width: calc(90% - 1rem);
  }
}

<div>
  <details
    class="emotion-0"
  >
    <summary
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        height="12"
        viewBox="0 0 32 32"
        width="12"
      >
        <g>
          <path
            d="M12.6,26.7L23.2,16L12.6,5.3H8.8v21.4H12.6z"
          />
        </g>
      </svg>
      <span
        role="text"
      >
        <span
          class="emotion-2"
        >
          Read transcript
        </span>
        <span
          class="emotion-3"
        >
          , My Title
        </span>
      </span>
    </summary>
    <small
      class="emotion-4"
    >
      This transcript has been reviewed by a journalist, it was generated with AI (artificial intelligence).
    </small>
    <ul
      class="emotion-5"
      role="list"
    >
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:00
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Na nutsu.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:01
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Ni 19 ne kuma wannan ita ce motata tafarko da na saya da kaina.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:04
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Na 1938 Austin goma Cambridge.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:14
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            A koyaushe ina sha'awar tarihin cewa akwaia cikin daji.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:18
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Don haka ba shakka, lokacin da nake samunkatina, dole ne ya zama tsohon.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:21
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Ba zai taba zama motar zamani ba.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:24
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            1112 shine lokacin da na yanke shawararcewa ina son daya da gaske kuma zan fara
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:28
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            tarawa ɗaya.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:29
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Don haka akwai 'yan kudin aljihu daabubuwa makamantansu.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:31
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Ƙananan ayyuka.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:32
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Zan sake yin ɗan kuɗin aljihu, ajiye shi.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:35
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Sannan a lokacin da nake makarantarsakandare, na samu aikin wucin gadi na,
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:39
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            sai wani bangare na albashina ya tafi,wanda hakan ya taimaka mini da sauri.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:43
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Ya kasance matashi sosai kuma yana tare dakakan da kaina da yawa.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:49
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Kuma um, sun kasance suna son, um,tsofaffin jiragen kasa na tururi da, um,
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            00:55
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            John yana da wasu motoci da ƙananan motocikuma ya kira su suna son duk tsofaffin
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:01
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            masu salo, wanda muke tunanin yana da bandariya sosai.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:06
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Wannan shine ainihin daftari da na samutare da wanda zai zo da kati a lokacin.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:12
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Kudin can £215, shilling 16 da £11 kenan.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:21
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Dama.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:21
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Kuna da abubuwa kamar haruffan zirga-zirga.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:25
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Akwai a cikin allonku ɗaya, gidaje.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:31
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Kamar da yawa.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:33
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Na tabbata zan kasance kyakkyawa a ranarzafi mai zafi.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:35
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Ban sami wannan damar ba tukuna, kodayake.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:37
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Kuma akwai anti dazzle kuma.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:40
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Wannan saitin a bayyane yake a cikinwannan motar.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:42
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Yin tuƙi a ciki, da gaske kun sami ma'anartarihi fiye da yadda za ku taɓa samu.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:47
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Tsaya ka gan su a tsaye a cikin gidankayan gargajiya kuma ina nufin da yawa
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:50
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            daga cikin wadannan motoci da gidajentarihi da abin da ba, ba zan sake komawa.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:53
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Su kenan har karshen rayuwarsu da gidankayan gargajiya.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:57
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Kuma ba abin da suke can ba ne.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            01:58
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Ana son a yi amfani da su kuma a more su.
          </span>
        </span>
      </li>
      <li
        class="emotion-6"
      >
        <span
          class="emotion-7"
          role="text"
        >
          <span
            class="emotion-8"
          >
            02:00
          </span>
          <span
            class="emotion-3"
          >
             
          </span>
          <span
            class="emotion-10"
          >
            Don haka, eh, abin da nake yi ke nan.
          </span>
        </span>
      </li>
    </ul>
  </details>
</div>
`;
